hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -274,9 +275,14 @@ public HoodieCompactionPlan scheduleCompaction(JavaSparkContext jsc, String inst
           return hoodieRollbackStats;
         }).collect(Collectors.toList())).flatMap(List::iterator).filter(Objects::nonNull).collect();
 
-    commitsAndCompactions.entrySet().stream().map(
-        entry -> new HoodieInstant(true, entry.getValue().getAction(),
-            entry.getValue().getTimestamp())).forEach(this.getActiveTimeline()::deleteInflight);
+    Stream<HoodieInstant> instantsToBeDeleted = commitsAndCompactions.entrySet().stream().map(
+        entry -> new HoodieInstant(true, entry.getValue().getAction(), entry.getValue().getTimestamp()));
+    if (deleteInstants) {
```

Comment:
nit: can lines 280-285 be shared between the classes?

Reply:
Fixed.